src/components/settings/SettingsLMSTab/ErrorReporting/DownloadCsvButton.jsx

```diff
@@ -0,0 +1,93 @@
+import React, { useState, useEffect } from 'react';
+import { saveAs } from 'file-saver';
+import PropTypes from 'prop-types';
+import {
+  Toast, StatefulButton, Icon, Spinner, useToggle,
+} from '@edx/paragon';
+import { Download, Check } from '@edx/paragon/icons';
+
+const DownloadCsvButton = ({ data }) => {
+  const [buttonState, setButtonState] = useState('pageLoading');
+
+  useEffect(() => {
+    if (data !== null && data.length !== 0) {
+      setButtonState('default');
+    }
+  }, [data]);
+
+  const getCsvFileName = () => {
+    const currentDate = new Date();
+    const year = currentDate.getUTCFullYear();
+    const month = currentDate.getUTCMonth() + 1;
+    const day = currentDate.getUTCDate();
+    return `${year}-${month}-${day}-error-log.csv`;
+  };
+
+  const getCsvData = () => {
+    if (!Array.isArray(data) || !data.every((p) => typeof p === 'object' && p !== null)) {
+      return null;
+    }
+    const heading = Object.keys(data[0]).join(',');
```

Comment:
Right now the heading is dynamic and not "pretty" because we assume staff might actually want to use the variable names.